src/addBusinessDays/index.ts

```diff
@@ -42,49 +42,64 @@ export default function addBusinessDays(
       ? [1, 2, 3, 4, 5]
       : options.businessDays.filter((number) => number < 7).map(toInteger)
 
-  const date = toDate(dirtyDate)
+  const initialDate = toDate(dirtyDate)
   const amount = toInteger(dirtyAmount)
   let accountedForExceptions = 0
+
   const isExcepted = (date: Date): boolean | null => {
     if (options.exceptions) {
       const exception =
         Object.keys(options.exceptions).find((e) =>
           isSameDay(new Date(e), date)
         ) || ''
-      if (options.exceptions[exception] !== undefined) {
+
+      if (options.exceptions[exception] === true) {
+        accountedForExceptions -= 1
+      } else if (options.exceptions[exception] === false) {
         accountedForExceptions += 1
       }
       return options.exceptions[exception]
     }
     return null
   }
-  const isNonWorkingDay = (date: Date) =>
-    isExcepted(date) === false || !businessDays.includes(date.getDay())
+
+  const isWorkingDay = (date: Date) => {
+    const hasException = isExcepted(date)
+    if (hasException === false) return false
+    if (hasException === true || businessDays.includes(date.getDay())) {
+      return true
+    }
+    return false
+  }
 
   if (isNaN(amount)) return new Date(NaN)
 
-  if (date.toString() === 'Invalid Date') {
-    return date
+  if (initialDate.toString() === 'Invalid Date') {
+    return initialDate
   }
-  const startedOnNonWorkingDay = isNonWorkingDay(date)
-  const hours = date.getHours()
+
+  const startedOnNonWorkingDay = !isWorkingDay(initialDate)
+  const hours = initialDate.getHours()
   const sign = amount < 0 ? -1 : 1
   const fullWeeks = toInteger(amount / businessDays.length)
-  date.setDate(date.getDate() + fullWeeks * 7)
+
+  let newDate = initialDate
```

Comment:
I changed the variable names because I found it a little confusing for everything to be called `date`